staging/src/k8s.io/apiserver/pkg/endpoints/request/context.go

```diff
@@ -88,6 +88,11 @@ func WithAuditEvent(parent context.Context, ev *audit.Event) context.Context {
 
 // AuditEventFrom returns the audit event struct on the ctx
 func AuditEventFrom(ctx context.Context) *audit.Event {
-	ev, _ := ctx.Value(auditKey).(*audit.Event)
```

Comment:
this being nil is concerning... is the audit filter at the wrong spot in the chain? cc @sttts

Reply:
the audit event get from context  can not be nil, it is guaranteed by 

```
func createAuditEventAndAttachToContext(req *http.Request, policy policy.Checker) (*http.Request, *auditinternal.Event, []auditinternal.Stage, error) {
	ctx := req.Context()
        .............

	req = req.WithContext(request.WithAuditEvent(ctx, ev))

	return req, ev, omitStages, nil
}
```

Reply:
@liggitt @wgliang @nikhita Accordingly with the stack trace I believe it's the `ctx` that is nil, not the audit itself.

```
pkg/endpoints/request.AuditEventFrom(0x0, 0x0, 0xc4255fc740)
                                     ^    ^
```

Reply:
> Accordingly with the stack trace I believe it's the ctx that is nil, not the audit itself.

makes sense to me.

Reply:
@aanm cool, but a check on audit should be necessary. Also, how does ctx happen to nil?

Reply:
The ctx could have been nil in former times before https://github.com/kubernetes/kubernetes/pull/62810 if there were two RequestContextMappers involved (only one got the context for the request).

Do we have any proof that this is still happening in 1.10+ ?